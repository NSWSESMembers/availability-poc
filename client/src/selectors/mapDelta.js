/*
  helpers for determining lat and long deltas when given lat + long and a distance.
  will return location with deltas so the map will render at a zoom level that will
  cover the given distance
*/

// eslint-disable-next-line import/prefer-default-export
export const getMapDelta = (lat, lon, distance) => {
  const oneDegreeOfLatitudeInMeters = 111.32 * 1000;

  const latitudeDelta = distance / oneDegreeOfLatitudeInMeters;
  const longitudeDelta = distance / (oneDegreeOfLatitudeInMeters * Math.cos(lat * (Math.PI / 180)));

  return {
    latitude: lat,
    longitude: lon,
    latitudeDelta,
    longitudeDelta,
  };
};
