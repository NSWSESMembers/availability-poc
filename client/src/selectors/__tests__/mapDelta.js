import { getMapDelta } from '../mapDelta';

test('mapDelta', () => {
  expect(
    getMapDelta(0, 0, 1),
  ).toMatchSnapshot();
});
