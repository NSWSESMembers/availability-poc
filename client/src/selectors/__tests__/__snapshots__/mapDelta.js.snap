// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapDelta 1`] = `
Object {
  "latitude": 0,
  "latitudeDelta": 0.00000898311174991017,
  "longitude": 0,
  "longitudeDelta": 0.00000898311174991017,
}
`;
